File: Python/String-1/the_end.py

# given a string, return a string length 1 from its front, unless front is false, in which case return a string length 1 from its back
# the string will be non-empty
def the_end(str, front):
    return str[:1] if front else str[len(str)-1:]

print(the_end('Hello', True))
print(the_end('Hello', False))
print(the_end('oh', True))

File: JavaScript/String-1/theEnd.js

// given a string, return a string length 1 from its front, unless front is false, in which case return a string length 1 from its back
// the string will be non-empty
const theEnd = (str, front) => {
    return front ? str[0] : str[str.length - 1];
};

console.log(theEnd('Hello', true));
console.log(theEnd('Hello', false));
console.log(theEnd('oh', true));
